### PR TITLE
TF-529: Add typing for `connect` call with bootstrap capability

### PR DESCRIPTION
**Pull request by p--b**
Previously, the typescript typing did not support calling `connect()` with a bootstrap capability. This commit resolves this. It is acknowledged that the typing provided does
not make many guarantees.

### CODE REVIEW

File: src/node-capnp/capnp.d.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ declare module Capnp {
   function parse<Builder, Reader>(type: StructSchema<Builder, Reader>, buffer: Buffer): Reader;
   function serialize<Builder, Reader>(type: StructSchema<Builder, Reader>, builder: Builder): Buffer;
   function connect(addr: string): Connection;
+  function connect<Client>(addr: string, clientBootstrapCapability: Client): Connection;
   function importFile<SchemaBag>(path: string): SchemaBag;
   function importSystem<SchemaBag>(path: string): SchemaBag;
```
Comment (rgprog) [inline src/node-capnp/capnp.d.ts:46]: Is there any advantage to using a generic parameter here, rather than typing `clientBootstrapCapability` as `object`?

Scope creep: I've felt similarly about `importFile` and `importSystem` for a while. The problem is more severe in their case, because their return value will implicitly cast to any type, when they could force the caller to write an explicit type assertion instead.

**Review by rgprog — APPROVED** [review]
Seems sensible 👍

File: src/node-capnp/capnp.d.ts (hunk 1 above)
Comment (p--b) [reply src/node-capnp/capnp.d.ts:46]: Only real advantage in my mind is that it's slightly more self-documenting.
`importFile` and `importSystem` are also fairly light touch - if you take a peek at how we use them they don't cast their return to `any` implicitly I don't believe.

Agreed it could all be improved with more time...
